From 74aebc79506b434543ae5e425075d25f548792a6 Mon Sep 17 00:00:00 2001
From: Razvan Stefanescu <razvan.stefanescu@nxp.com>
Date: Mon, 30 Apr 2018 11:27:27 +0300
Subject: [PATCH 525/706] staging: dpaa2-evb: Defer probe if no mc portal is
 found

Signed-off-by: Razvan Stefanescu <razvan.stefanescu@nxp.com>
(cherry picked from commit 5bba03126bbbef26b7775f9fbd4ef12859bae504)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/staging/fsl-dpaa2/evb/evb.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/evb/evb.c b/drivers/staging/fsl-dpaa2/evb/evb.c
index 86037640a211..cdce6144ed8b 100644
--- a/drivers/staging/fsl-dpaa2/evb/evb.c
+++ b/drivers/staging/fsl-dpaa2/evb/evb.c
@@ -1204,10 +1204,14 @@ static int evb_probe(struct fsl_mc_device *evb_dev)
 
 	err = fsl_mc_portal_allocate(evb_dev, FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
 				     &priv->mc_io);
-	if (unlikely(err)) {
-		dev_err(dev, "fsl_mc_portal_allocate err %d\n", err);
+	if (err) {
+		if (err == -ENXIO)
+			err = -EPROBE_DEFER;
+		else
+			dev_err(dev, "fsl_mc_portal_allocate err %d\n", err);
 		goto err_free_netdev;
 	}
+
 	if (!priv->mc_io) {
 		dev_err(dev, "fsl_mc_portal_allocate returned null handle but no error\n");
 		err = -EFAULT;
-- 
2.17.1

